### DISCUSSION TITLE
Articles types schema

### DISCUSSION THREAD

**Discussion by atherdon** [atherdon/dash-back#52] [General]
Т.к. мы уже решили кучу важных вопросов - и фронтенд немного отстает со своими апдейтами - можно подумать о создании нормальной схемы, в которой появятся связи между таблицами. Посути это уже работа на будущее.
 
Мы можем делать эту часть сейчас, но я хочу сфокусироваться на запуске текущей версии - чтобы редакторы и Марина уже смогли начать использовать нашу админку и начали давать свой feedback.

Prisma хороша тем, что миграции должны проходить без проблем, поэтому мы сможем добавлять наши улучшения налету. Я позже расширю эту схему и добавлю еще пару таблиц и связей, о которых забыл.
Я решил нарисовать эту схему потому что в последнем PR я увидел [схему Articles](https://github.com/atherdon/dash-back/blob/main/src/types/graphql/index.d.ts#L35) и понял что мы можем развивать эту часть дальше. 

![Articles schema](https://user-images.githubusercontent.com/1469198/108333009-1fc86200-71d9-11eb-99ad-5423a2eea853.jpg)

### Bad Keywords

сейчас это просто 
- id
- String
/ будем выводить списком на фронте, чтобы редакторы не забывали о них.

### Homepage Keeper

- id 
- Category 
- Number 
/ хочется иметь тулзу, которая будет показывать редакторам, какие категории на главной странице нуждаются в новых статьях.

![dashb2](https://user-images.githubusercontent.com/1469198/108334369-8a2dd200-71da-11eb-8d5c-ae66352f9d23.png)

### Stages\Statuses

наверное лучше это делать отдельным milestone - я пока неготов обсуждать эту часть. мне нужно разобраться в моих бумагах и найти описание, которое я делал в июле.

### Authors

необязательная часть, но рано или поздно мы прийдем к этому. статьи так или иначе привязаны к автору. Если наши программисты помогут делать экспорт - мы сможем и авторов экспортировать для удобства. В близжайщие 6 недель я не планирую эту часть проекта трогать, но начало можно положить.

### Managed Accounts, Fellowship, Top Authors

это одновременно и тип для статьи, и тип автора.
тоесть у нас есть набор авторов, которые мы сами менеджерим и выкладываем. 
Чтобы нам было проще, хочется разделить статьи на разные "кучки", чтобы редакторам было проще 

### Evergreen aka Brands

мы пытаемся монетизировать эту часть нашей работы, поэтому можно будет детальнее обсудить эту секцию.
проблема которая есть сейчас: ничего не автоматизированно, плюс в одной "куче" могут лежать статьи, за которые заплатили и статьи из freemium программы(мы дарим 3 статьи тем кто регает новый аккаунт). сейчас это все смешивается.
тоесть если даже просто понимать - относится статья к freemium или платная - это уже будет хорошее улучшение.
наверное будет связка с автором\брендом, чтобы мы могли видеть больше информации - перед тем как откроем саму статью.
Хотелось бы добавлять notes к автору. и main category в которой выкладываются статьи. 

### Slogging

эти статьи сейчас имеют плохой формат. тоесть чтобы статьи этого типа выложить - нужно сделать больше подготовительной работы. поэтому я вообще хотел бы чтобы отдельный junior вначале форматировал эти статьи, а потом они уже прилетали к редактору на просмотр.

### Ordinary articles

наверное этот тип мы будем использовать, когда с основного проекта будет приходить экспорт информации. Поэтому автоматически многие статьи будут marked as ordinary - а уже потом Марина или редактора будут менять тип статьи.

### New submissions
очень часто бывает такая ситуация: ктото продал копирайт статью у нас на сайте ради SEO ссылки. И этот человек пытается всеми способами ее выложить. создает фейковые аккаунты и т.д. Если мы введем данный тип - то в него будут попадать все неблагонадежные статьи. и эти статьи non visible основным редакторам. 
Марина вначале будет их проверять, и если статья плохая или например содержит **bad keywords** - отклонять(status : rejected).
Если статья ок - будет менятся тип и после этого, данная статья будет visible основным редакторам.

### Spam?

должен ли у нас быть тип спам? сейчас неуверен, но нам прилетает много плохих статей. возможно спам это не тип статьи а скорее статус или stage.

### Top Contributors
на первый взгляд этот тип похож на top authors. но мы хотим сделать так, чтобы крутые авторы могли сами выкладывать свои статьи. и этот тип статьи помог бы убирать их из нашего основного queue в будущем. 

---

когда схема будет обновлена, мы можем добавить работу с типами на [dashboard-upload-forms](https://github.com/atherdon/dash-upload-forms) потому что Марина будет первым человеком, который будет фильтровать статьи по типам и добавлять новые данные в базу.
А когда эта часть уже будет обкатана - мы сможем перенести этот функционал и в основной dashboard

**Comment by kolserdav** [comment]
> Prisma хороша тем, что миграции должны проходить без проблем, поэтому мы сможем добавлять наши улучшения налету. Я позже расширю эту схему и добавлю еще
>
По идее да, если нужно добавить колонку например, но если поменять тип колонки в которой уже есть данные, то скорее всего не даст и придётся перегружать как-то.
